Platform/OcTextInputDxe/OcTextInputDxe.c

```
@@ -0,0 +1,867 @@
+{
+  DEBUG ((
+    DEBUG_INFO,
+    "OcTextInputDxe: System is ready to boot, performing late initialization\n"
+    ));
+  PerformCompatibilityInstallation ();
+}
+
+/**
+   The entry point for OcTextInputDxe driver.
+
+   @param[in] ImageHandle    The firmware allocated handle for the EFI image.
+   @param[in] SystemTable    A pointer to the EFI System Table.
+
+   @retval EFI_SUCCESS       The entry point is executed successfully.
+   @retval other             Some error occurs when executing this entry point.
+ **/
+EFI_STATUS
+EFIAPI
+OcTextInputDxeEntry (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+
+  DEBUG ((
+    DEBUG_INFO,
+    "OcTextInputDxe: === OpenCore SimpleTextInputEx Compatibility Driver Starting ===\n"
+    ));
+  DEBUG ((
+    DEBUG_INFO,
+    "OcTextInputDxe: System Table Revision: 0x%08X (%d.%02d)\n",
+    SystemTable->Hdr.Revision,
+    SystemTable->Hdr.Revision >> 16,
+    SystemTable->Hdr.Revision & 0xFFFF
+    ));
+
+  if (SystemTable->RuntimeServices != NULL) {
+    DEBUG ((
+      DEBUG_INFO,
+      "OcTextInputDxe: Runtime Services Revision: 0x%08X (%d.%02d)\n",
+      SystemTable->RuntimeServices->Hdr.Revision,
+      SystemTable->RuntimeServices->Hdr.Revision >> 16,
+      SystemTable->RuntimeServices->Hdr.Revision & 0xFFFF
+      ));
+  }
+
+  if (SystemTable->BootServices != NULL) {
+    DEBUG ((
+      DEBUG_INFO,
+      "OcTextInputDxe: Boot Services Revision: 0x%08X (%d.%02d)\n",
+      SystemTable->BootServices->Hdr.Revision,
+      SystemTable->BootServices->Hdr.Revision >> 16,
+      SystemTable->BootServices->Hdr.Revision & 0xFFFF
+      ));
+  }
+
+  // Check if console services are available for immediate installation
+  if ((SystemTable->ConsoleInHandle != NULL) && (SystemTable->ConIn != NULL)) {
+    DEBUG ((DEBUG_INFO, "OcTextInputDxe: Console services are available, attempting immediate installation\n"));
+    Status = PerformCompatibilityInstallation ();
```

The implementation of PerformCompatibilityInstallation swallows it's errors and always returns EFI_SUCCESS.